plugin/pkg/global/extensionlabels/admission.go

```diff
@@ -0,0 +1,256 @@
+
+	case core.Kind("BackupEntry"):
+		backupEntry, ok := a.GetObject().(*core.BackupEntry)
+		if !ok {
+			return apierrors.NewBadRequest("could not convert object into BackupEntry object")
+		}
+
+		backupBucket, err := e.backupBucketLister.Get(backupEntry.Spec.BucketName)
+		if err != nil {
+			return err
+		}
+
+		removeLabels(&backupEntry.ObjectMeta)
+		addMetaDataLabelsBackupEntry(backupEntry, backupBucket)
+	}
+	return nil
+}
+
+func addMetaDataLabelsSeed(seed *core.Seed) {
+	metav1.SetMetaDataLabel(&seed.ObjectMeta, v1beta1constants.LabelExtensionProviderPrefix+seed.Spec.Provider.Type, "true")
+	if seed.Spec.Backup != nil {
+		metav1.SetMetaDataLabel(&seed.ObjectMeta, v1beta1constants.LabelExtensionProviderPrefix+seed.Spec.Backup.Provider, "true")
+	}
+
+	if seed.Spec.DNS.Provider != nil && utilfeature.DefaultFeatureGate.Enabled(features.UseDNSRecords) {
+		metav1.SetMetaDataLabel(&seed.ObjectMeta, v1beta1constants.LabelExtensionDNSRecordPrefix+seed.Spec.DNS.Provider.Type, "true")
+	}
+}
+
+func addMetaDataLabelsShoot(shoot *core.Shoot) {
+	for _, extension := range shoot.Spec.Extensions {
+		if extension.Disabled != nil && *extension.Disabled {
+			continue
+		}
+		metav1.SetMetaDataLabel(&shoot.ObjectMeta, v1beta1constants.LabelExtensionTypePrefix+extension.Type, "true")
+	}
+	for _, pool := range shoot.Spec.Provider.Workers {
+		if pool.CRI != nil {
+			for _, cr := range pool.CRI.ContainerRuntimes {
+				metav1.SetMetaDataLabel(&shoot.ObjectMeta, v1beta1constants.LabelExtensionContainerRuntimePrefix+cr.Type, "true")
+			}
+		}
+		if pool.Machine.Image != nil {
+			metav1.SetMetaDataLabel(&shoot.ObjectMeta, v1beta1constants.LabelExtensionOperatingSystemConfigPrefix+pool.Machine.Image.Name, "true")
+		}
+	}
+	if shoot.Spec.DNS != nil {
+		for _, provider := range shoot.Spec.DNS.Providers {
+			if provider.Type == nil || *provider.Type == core.DNSUnmanaged {
+				continue
+			}
+			if utilfeature.DefaultFeatureGate.Enabled(features.UseDNSRecords) {
+				metav1.SetMetaDataLabel(&shoot.ObjectMeta, v1beta1constants.LabelExtensionDNSRecordPrefix+*provider.Type, "true")
+			}
+		}
+	}
+	metav1.SetMetaDataLabel(&shoot.ObjectMeta, v1beta1constants.LabelExtensionProviderPrefix+shoot.Spec.Provider.Type, "true")
+	metav1.SetMetaDataLabel(&shoot.ObjectMeta, v1beta1constants.LabelExtensionNetworkingPrefix+shoot.Spec.Networking.Type, "true")
+}
+
+func addMetaDataLabelsCloudProfile(cloudProfile *core.CloudProfile) {
+	for _, machineImage := range cloudProfile.Spec.MachineImages {
+		metav1.SetMetaDataLabel(&cloudProfile.ObjectMeta, v1beta1constants.LabelExtensionOperatingSystemConfigPrefix+machineImage.Name, "true")
+
+		for _, machineImageVersion := range machineImage.Versions {
+			for _, cri := range machineImageVersion.CRI {
+				for _, cr := range cri.ContainerRuntimes {
+					metav1.SetMetaDataLabel(&cloudProfile.ObjectMeta, v1beta1constants.LabelExtensionContainerRuntimePrefix+cr.Type, "true")
+				}
+			}
+		}
+	}
```

Comment:
I think this is not needed because OSC/CR extensions don't act on `CloudProfile`s (there is not even `providerConfig` for them). Just adding the provider type label for `CloudProfile`s is enough.